app/src/main/java/com/monkeyteam/chimpagne/ui/components/Button.kt

```diff
@@ -97,3 +108,50 @@ fun GoBackButton(navigationActions: NavigationActions) {
         tint = MaterialTheme.colorScheme.onSurface)
   }
 }
+
+@Composable
+fun CalendarButton(event: ChimpagneEvent?, contextMainActivity: Context) {
+  var showDialog by remember { mutableStateOf(false) }
+  IconButton(modifier = Modifier.testTag("calendarButton"), onClick = { showDialog = true }) {
+    Icon(
+        imageVector = Icons.Default.CalendarMonth,
+        contentDescription = "Add to Calendar",
+        tint = MaterialTheme.colorScheme.onSurface,
+        modifier = Modifier.testTag("calendarButtonIcon"))
+  }
+  if (showDialog) {
+    popUpCalendar(
+        onAccept = {
+          val intent = createCalendarIntent(event)
+          if (intent != null) {
+            contextMainActivity.startActivity(intent)
+          } else {
+            Toast.makeText(
+                    contextMainActivity, "Event can't be added to calendar", Toast.LENGTH_SHORT)
+                .show()
+          }
+          showDialog = false
+        },
+        onReject = { showDialog = false },
+        event = event)
+  }
+}
+
+@Composable
+fun popUpCalendar(onAccept: () -> Unit, onReject: () -> Unit, event: ChimpagneEvent?) {
+  val builder =
+      AlertDialog(
+          onDismissRequest = { onReject() },
+          title = { Text("Add to Calendar") },
+          text = { Text("Do you want to add the event \"${event?.title}\" to your calendar?") },
```

Comment:
Here as well, make it as a StringResource

Reply:
done